cfme/tests/infrastructure/test_kubevirt.py

```diff
@@ -3,23 +3,94 @@
+    template_name = provisioning['template']
+    vm = appliance.collections.infra_vms.instantiate(name=vm_name,
+                                                     provider=provider,
+                                                     template_name=template_name)
+    note = ('template {} to vm {} on provider {}'.format(template_name, vm_name, provider.key))
+    prov_data.update({
+        'request': {
+            'email': 'template_provisioner@example.com',
+            'first_name': 'Template',
+            'last_name': 'Provisioner',
+            'notes': note}})
+    view = navigate_to(vm, 'Provision')
+    view.form.fill_with(prov_data, on_change=view.form.submit_button)
+    view.flash.assert_no_error()
+    vm.wait_to_appear()
+    yield vm
+    # vm.retire()
+    # DO NOT retire VM for now on kubevirt becuase of Kernel panic issue
+
+
+def test_k6t_provider_crud(provider):
 
     with update(provider):
         provider.name = fauxfactory.gen_alphanumeric() + '_updated'
 
     provider.delete(cancel=False)
     provider.wait_for_delete()
+
+
+@pytest.mark.parametrize('custom_prov_data',
+                         [{}, {'hardware': {'cpu_cores': '8', 'memory': '8192'}}],
+                         ids=['via_lifecycle', 'override_template_values'])
+def test_k6t_vm_crud(request, appliance, provider, vm_name, provisioning,
+                     prov_data, custom_prov_data):
+    provider.refresh_provider_relationships()
+    prov_data.update(custom_prov_data)
+    template = provisioning['template']
+    do_vm_provisioning(appliance, template, provider, vm_name, prov_data, request,
+                       smtp_test=False, wait=False)
+    logger.info('Waiting for cfme provision request for vm %s', vm_name)
+    request_description = 'Provision from [{}] to [{}]'.format(template, vm_name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui', num_sec=300)
+    assert provision_request.is_succeeded(method='ui'), \
+        ("Provisioning failed with the message {}".format(provision_request.row.last_message.text))
+
+
+@pytest.mark.parametrize(['power_option', 'vm_state'],
+                         [('Power On', 'currentstate-on'), ('Power Off', 'currentstate-off')])
+def test_vm_power_management(request, appliance, provider, temp_vm,
+                             power_option, vm_state):
+    temp_vm.power_control_from_cfme(option=power_option)
+    provider.refresh_provider_relationships()
+    # to refresh provider and wait for power states to get updated
+    assert vm_state in temp_vm.find_quadicon().data['state']
+
+
+def test_vm_power_management_from_details(request, appliance, provider, temp_vm):
+    temp_vm.power_control_from_cfme(from_details=True, option='Power Off')
+    provider.refresh_provider_relationships()
+    assert 'currentstate-off' in temp_vm.find_quadicon().data['state']
```

Comment:
Why not parametrize a single test function on both `('power_option', 'vm_state')` and `('from_collection', 'from_details')`.

Additionally I think it would be good to pass an explicit `ids` to both of these parametrization calls to be explicit.  Most 'from_details' type parametrization is done as a boolean, with specific ID's.

Reply:
changed